### PR TITLE
Add ESInputTag to VPSet default added to cfi file

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

Also added unit test for cfi content of VPSet.

#### PR validation:

Code compiles and new unit test passes.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: FWCore/ParameterSet/src/FillDescriptionFromPSet.cc [hunk 1]
```diff
@@ -40,41 +41,34 @@ namespace {
...
-    s_findTheRightFunction[static_cast<edm::ParameterTypes>('t')] = &fillDescriptionFromParameter<edm::InputTag>;
-    s_findTheRightFunction[static_cast<edm::ParameterTypes>('v')] =
-        &fillDescriptionFromParameter<std::vector<edm::InputTag>>;
-    s_findTheRightFunction[static_cast<edm::ParameterTypes>('F')] = &fillDescriptionFromParameter<edm::FileInPath>;
-    s_findTheRightFunction[static_cast<edm::ParameterTypes>('A')] =
-        &fillDescriptionFromParameter<edm::LuminosityBlockRange>;
-    s_findTheRightFunction[static_cast<edm::ParameterTypes>('a')] =
+    s_findTheRightFunction[edm::k_InputTag] = &fillDescriptionFromParameter<edm::InputTag>;
+    s_findTheRightFunction[edm::k_VInputTag] = &fillDescriptionFromParameter<std::vector<edm::InputTag>>;
+    s_findTheRightFunction[edm::k_ESInputTag] = &fillDescriptionFromParameter<edm::ESInputTag>;
```
Comment (Dr15Jones) [inline FWCore/ParameterSet/src/FillDescriptionFromPSet.cc:64]: These were the two missing entries.

The rest should just be a direct replacement of the char with the same valued enum.

File: FWCore/ParameterSet/src/FillDescriptionFromPSet.cc [hunk 2]
```diff
@@ -40,41 +41,34 @@ namespace {
...
+    s_findTheRightFunction[edm::k_VESInputTag] = &fillDescriptionFromParameter<std::vector<edm::ESInputTag>>;
```
Comment (Dr15Jones) [inline FWCore/ParameterSet/src/FillDescriptionFromPSet.cc:65]: These were the two missing entries.

The rest should just be a direct replacement of the char with the same valued enum.

File: FWCore/ParameterSet/test/test_catch_ParameterSetDescription.cc [hunk 3]
```diff
@@ -1252,52 +1252,173 @@ TEST_CASE("test ParameterSetDescription", "[ParameterSetDescription]") {
...
+    vLr = cms.VLuminosityBlockRange('1:2-3:4'),
+    vd = cms.vdouble(1),
+    ve = cms.VEventID('1:2:3'),
+    ver = cms.VEventRange('1:2:3-4:5:6'),
+    vet = cms.VESInputTag(':foo'),
+    vi = cms.vint32(1),
+    vl = cms.vint64(1),
+    vt = cms.VInputTag('foo'),
+    vui = cms.vuint32(1),
+    vul = cms.vuint64(1)
```
Comment (makortel) [inline FWCore/ParameterSet/test/test_catch_ParameterSetDescription.cc:1368]: I see `string`, `vstring`, `FileInPath`, `vbool` are "missing" from this list. Are they tested well-enough elsewhere, or would there be value in adding them here?

(in principle also inner `PSet` and `VPSet` are not there, but maybe they are tested already well-enough elsewhere?)

Comment (Dr15Jones) [reply FWCore/ParameterSet/test/test_catch_ParameterSetDescription.cc:1368]: We don't have a `vbool` in the system.  Seems FileInPath's can't be created without the file being there (there is a special test in this unit test working around that).  I'll definitely add the `string` and `vstring`.

Comment (Dr15Jones) [reply FWCore/ParameterSet/test/test_catch_ParameterSetDescription.cc]: I was able to add all the presently available options to the test.

**Comment by Dr15Jones** [comment]
please test

**Comment by makortel** [comment]
+core

**Comment by antoniovilela** [comment]
+1

**Comment by Martin-Grunewald** [comment]
In the most recent relvals (CMSSW_14_1_X_2024-03-10-0000), there are many errors of the type:
```
Traceback (most recent call last):
  File "/cvmfs/cms-ib.cern.ch/sw/x86_64/week0/el8_amd64_gcc12/cms/cmssw/CMSSW_14_1_X_2024-03-10-0000/bin/el8_amd64_gcc12/cmsDriver.py", line 40, in <module>
    run()
  File "/cvmfs/cms-ib.cern.ch/sw/x86_64/week0/el8_amd64_gcc12/cms/cmssw/CMSSW_14_1_X_2024-03-10-0000/bin/el8_amd64_gcc12/cmsDriver.py", line 16, in run
    configBuilder.prepare()
  File "/cvmfs/cms-ib.cern.ch/sw/x86_64/week0/el8_amd64_gcc12/cms/cmssw/CMSSW_14_1_X_2024-03-10-0000/src/Configuration/Applications/python/ConfigBuilder.py", line 2284, in prepare
    self.loadAndRemember(self.EVTCONTDefaultCFF)  #load the event contents regardless
  File "/cvmfs/cms-ib.cern.ch/sw/x86_64/week0/el8_amd64_gcc12/cms/cmssw/CMSSW_14_1_X_2024-03-10-0000/src/Configuration/Applications/python/ConfigBuilder.py", line 354, in loadAndRemember
    self.process.load(includeFile)
  File "/cvmfs/cms-ib.cern.ch/sw/x86_64/week0/el8_amd64_gcc12/cms/cmssw/CMSSW_14_1_X_2024-03-10-0000/src/FWCore/ParameterSet/python/Config.py", line 761, in load
    module = __import__(moduleName)
  File "/cvmfs/cms-ib.cern.ch/sw/x86_64/week0/el8_amd64_gcc12/cms/cmssw/CMSSW_14_1_X_2024-03-10-0000/src/Configuration/EventContent/python/EventContent_cff.py", line 64, in <module>
    from RecoHGCal.Configuration.RecoHGCal_EventContent_cff import *
  File "/cvmfs/cms-ib.cern.ch/sw/x86_64/week0/el8_amd64_gcc12/cms/cmssw/CMSSW_14_1_X_2024-03-10-0000/src/RecoHGCal/Configuration/python/RecoHGCal_EventContent_cff.py", line 3, in <module>
    from RecoHGCal.TICL.iterativeTICL_cff import ticlIterLabelsMerge
  File "/cvmfs/cms-ib.cern.ch/sw/x86_64/week0/el8_amd64_gcc12/cms/cmssw/CMSSW_14_1_X_2024-03-10-0000/src/RecoHGCal/TICL/python/iterativeTICL_cff.py", line 16, in <module>
    from RecoHGCal.TICL.tracksterSelectionTf_cfi import *
  File "/cvmfs/cms-ib.cern.ch/sw/x86_64/week0/el8_amd64_gcc12/cms/cmssw/CMSSW_14_1_X_2024-03-10-0000/src/RecoHGCal/TICL/python/tracksterSelectionTf_cfi.py", line 1, in <module>
    from PhysicsTools.TensorFlow.tfGraphDefProducer_cfi import tfGraphDefProducer as _tfGraphDefProducer
  File "/cvmfs/cms-ib.cern.ch/sw/x86_64/week0/el8_amd64_gcc12/cms/cmssw/CMSSW_14_1_X_2024-03-10-0000/cfipython/el8_amd64_gcc12/PhysicsTools/TensorFlow/tfGraphDefProducer_cfi.py", line 5, in <module>
    FileName = cms.FileInPath(),
TypeError: __init__() missing 1 required positional argument: 'value'
```

It seems they originate from this PR which touches ```FileInPath``` though I do not understand why the PR tests of this PR went fine.

**Comment by Dr15Jones** [comment]
@Martin-Grunewald thanks for reporting the problem. I'll work on a fix today.

**Comment by Martin-Grunewald** [comment]
Fixed by #44362

**Comment by antoniovilela** [comment]
> Fixed by #44362

this is now merged

**Comment by antoniovilela** [comment]
> It seems they originate from this PR which touches `FileInPath` though I do not understand why the PR tests of this PR went fine.

It would be nice to understand why.

**Comment by Dr15Jones** [comment]
> It would be nice to understand why

This PR, I changed the 'empty' generated cfi version of FileInPath from `cms.FileInPath('')` to `cms.FileInPath()` which turned out not to parse properly in python.

**Comment by antoniovilela** [comment]
> > It would be nice to understand why
> 
> This PR, I changed the 'empty' generated cfi version of FileInPath from `cms.FileInPath('')` to `cms.FileInPath()` which turned out not to parse properly in python.

Thanks. But why nothing showed up in PR tests?

**Comment by Dr15Jones** [comment]
> Thanks. But why nothing showed up in PR tests?

That I do not know but would like to find out.

**Comment by makortel** [comment]
This PR touched only source files in `FWCore/ParameterSet` and `FWCore/Utilities`, and therefore not other packages (except `Utilities/RelMon` and `Utilities/StaticAnalyzer`, don't know why) were checked out in the PR tests, and therefore none of the tests that failed in IBs were run in the PR tests.

In retrospect we maybe should have tested this PR with `full_cmssw = true`.

**Pull request merged**